### PR TITLE
Backport #3115 create predecessors in generated js classes and deferred relationships fixes

### PR DESCRIPTION
**Pull request by MichaelBelousov**
- backport #3115 
- logical changes in the backport:
  - the new imports in the 3.0 version of this PR introduced a cyclic dependency when backporting to 2.19.x, I chose to fix this with a lazy deferred import instead of structural changes

### CODE REVIEW

**Comment by pmconne** [comment]
When you run rush change make sure to do `-b origin/release/2.19.x`

**Comment by calebmshafer** [comment]
@skirby1996 this should be included in a new patch soon. Probably should have gone out with the previous native release.

File: core/backend/src/ClassRegistry.ts [hunk 1]
```diff
@@ -6,11 +6,21 @@
...
+import { EntityMetaData, IModelError, RelatedElement } from "@bentley/imodeljs-common";
 import { Entity } from "./Entity";
 import { IModelDb } from "./IModelDb";
 import { Schema, Schemas } from "./Schema";
+import type { Element } from "./Element"; // type import to prevent cyclic dependency
+
+// the above "./Element" import causes a cyclic dependency in 2.19.x, but not 3.x
+// As such, the runtime module import is done lazily here to avoid the cyclic dependency without moving things around
+class LazilyImported {
+  private static _Element: Element | undefined;
```
Comment (MichaelBelousov) [inline core/backend/src/ClassRegistry.ts:19]: this is probably incorrect, the type should be `typeof Element`

Comment (MichaelBelousov) [reply core/backend/src/ClassRegistry.ts]: fixed and wrapped the require call in an `import()` type

**Comment by MichaelBelousov** [comment]
@skirby1996 @calebmshafer have you seen this issue before? https://dev.azure.com/bentleycs/iModelTechnologies/_build/results?buildId=1283509&view=logs&j=b471aebf-444e-5a10-bd73-fd9247c2fb87&t=854dcdcb-9d60-5f1d-54f5-e32095ba26f8&l=1041
just happened twice in a row for me

**Comment by calebmshafer** [comment]
> @skirby1996 @calebmshafer have you seen this issue before? https://dev.azure.com/bentleycs/iModelTechnologies/_build/results?buildId=1283509&view=logs&j=b471aebf-444e-5a10-bd73-fd9247c2fb87&t=854dcdcb-9d60-5f1d-54f5-e32095ba26f8&l=1041
> just happened twice in a row for me

@MichaelBelousov I've seen it happen a couple times randomly and seems to be always correlated with long install times. Haven't tracked it down further than that.

We are having an issue with some of the Mac CI machines due to the OS upgrades. We're looking into it and should hopefully be resolved by tomorrow.

In the meantime, mind just retrying?

**Comment by MichaelBelousov** [comment]
> In the meantime, mind just retrying?

Will do

**Comment by MichaelBelousov** [comment]
@calebmshafer going to delay this until the model json bug is fixed

**Comment by MichaelBelousov** [comment]
@calebmshafer the native backport is here https://dev.azure.com/bentleycs/iModelTechnologies/_git/imodel02/pullrequest/225083

**Comment by MichaelBelousov** [comment]
need to cherry-pick #3179 into this since that contains a fix for a js bug in the changes

**Comment by MichaelBelousov** [comment]
@calebmshafer when the new 2.x patch addon is released can you review these cherry-picks?

**Comment by skirby1996** [comment]
I've merged the new 2.x patch addon into this branch, seems like the builds are passing with it.

File: common/api/imodeljs-backend.api.md [hunk 2]
```diff
@@ -1859,6 +1859,9 @@ export class Entity implements EntityProps {
     forEachProperty(func: PropertyCallback, includeCustom?: boolean): void;
     id: Id64String;
     iModel: IModelDb;
+    static is(otherClass: typeof Entity): boolean;
```
Comment (calebmshafer) [inline common/api/imodeljs-backend.api.md:1862]: did you mean to make this `public`?

Comment (MichaelBelousov) [reply common/api/imodeljs-backend.api.md:1863]: made it beta for 2.x. Please resolve if it looks good

Comment (MichaelBelousov) [reply common/api/imodeljs-backend.api.md:1863]: hmmm I don't see the beta tag on this file, let me check

Comment (pmconne) [reply common/api/imodeljs-backend.api.md:1863]: Make sure you `npm run build` before `npm run extract-api`.

Comment (MichaelBelousov) [reply common/api/imodeljs-backend.api.md:1863]: I did it correctly originally, but github's UI didn't show me "outdated" or update my commit range in the "files" tab.
It's there, please review

**Review by calebmshafer — APPROVED** [review]

**Pull request merged**